src/Engine/StateTrackers/KillCountsTracker.java

```diff
@@ -0,0 +1,57 @@
+package Engine.StateTrackers;
+
+import Engine.GameInstance;
+import Engine.XYCoord;
+import Engine.Combat.BattleSummary;
+import Engine.GameEvents.GameEventQueue;
+import Engine.UnitActionLifecycles.JoinLifecycle.JoinEvent;
+import Units.Unit;
+
+/**
+ * Tracks a simple kill count for each unit; joined units use the higher value.
+ */
+public class KillCountsTracker extends StateTracker
+{
+  private static final long serialVersionUID = 1L;
+
+  // I wrote a generic handler for this and I'm going to use it
+  private CountManager<GameInstance, Unit> killCounts = new CountManager<>();
```

Comment:
Sigh. I think this class has been overkill everywhere I've actually seen it used. Cinder uses it, but only ever uses her own army as the list key. Now we have one where the GameInstance is the only key.

I mean, I guess it works.

Reply:
It is global per `GameInstance`, so you need *some* distinguishing factor... unless it really is global.
It might be doable to not require the first key.